safe-global/safe-dao-governance-app
# [EPIC] SEP #5 - Contract interaction

## What it solves

Resolves SEP #5 contract interaction

## How this PR fixes it

This integrates with a test airdrop contract `0xB1E81Fe4442FA5A0eE38feB7146E547937C6E737`, allocating `69` tokens to `0x5f310dc66F4ecDE9a1769f1B7D75224dA592201e`. All claims are batched with the test contract first to ensure redemptions for SEP #5.

The [above allocation is pushed to the current data locally](https://github.com/safe-global/safe-dao-governance-app/blob/2cb2b3f64e01f7d07ce623c74513a543fb963370/src/hooks/useAllocations.ts#L21-L32) if the Safe is being used with the app. It replaces all previously static mock data in the interface.

## How to test it

1. Open the `0x5f310dc66F4ecDE9a1769f1B7D75224dA592201e` Safe in two windows.
2. Add this PR deployment as a custom Safe App in one.
3. Add the dev deployment (`https://safe-dao-governance.dev.5afe.dev`) as a custom Safe App in the other.
4. Observe the higher allocation (of 69 value) in the PR deployment.
5. Claim the same amount of tokens in both windows and observe the PR deployment transaction modal showing the test contract above that of the old one.

## Screenshots

![image](https://github.com/safe-global/safe-dao-governance-app/assets/20442784/dc65b15f-633a-4714-91cd-54b230bf35e2)

## Checklist
